signal_scanner_bot/messages.py

```diff
@@ -51,6 +52,50 @@ def _condense_command(message: str) -> str:
     return NON_ALPHA_NUMERIC.sub("", message).upper()
 
 
+def _strip_tweet_hashtags(status_text: str) -> str:
+    """
+    Strip out words from tweet that are hashtags (ie. begin with a #)
+    """
+    text_split = [word for word in status_text.split() if not word.startswith("#")]
+    text = " ".join(text_split)
+    return text
+
+
+def get_tweet_text(status: Status) -> str:
+    """
+    Extract full text whether tweet is extended or not
+    """
+    if hasattr(status, "extended_tweet"):
+        return _strip_tweet_hashtags(status.extended_tweet["full_text"])
+    else:
+        return _strip_tweet_hashtags(status.text)
+
+
+def format_retweet_text(status: Status) -> str:
+    """
+    Extract text from retweet and build signal message
+    """
+    # Pull text out of the main tweet and sub tweet
+    top_level_tweet_text = get_tweet_text(status)
+    quoted_tweet_text = get_tweet_text(status.quoted_status)
+
+    # Build Signal message
+    if top_level_tweet_text:
+        return dedent(
+            f"""\
+        top level tweet:
```

Comment:
To differentiate this from the message context, it might be best to have this be in all caps, possibly with some character signifier?
```suggestion
        [QUOTE TWEET]:
```